randutil/random.go

```diff
@@ -22,80 +19,89 @@ func init() {
 }
 
 // Salt generates a new random salt of the given size.
-func Salt(size int) ([]byte, error) {
+func Salt(size int) []byte {
 	salt := make([]byte, size)
-	_, err := io.ReadFull(rand.Reader, salt)
-	if err != nil {
-		return nil, errors.Wrap(err, "error generating salt")
+	// crypto/rand.Reader always fill the entire slices.
+	// Since Go 1.24, it is guaranteed to never return an error, see https://github.com/golang/go/issues/66821.
+	// This error check can be removed once we drop support for Go 1.23.
+	if _, err := rand.Read(salt); err != nil {
+		panic("rand.Reader returned an error")
 	}
-	return salt, nil
+	return salt
 }
 
 // Bytes generates a new byte slice of the given size.
-func Bytes(size int) ([]byte, error) {
+func Bytes(size int) []byte {
 	bytes := make([]byte, size)
-	_, err := io.ReadFull(rand.Reader, bytes)
-	if err != nil {
-		return nil, errors.Wrap(err, "error generating bytes")
+	// crypto/rand.Reader always fill the entire slices.
+	// Since Go 1.24, it is guaranteed to never return an error, see https://github.com/golang/go/issues/66821.
+	// This error check can be removed once we drop support for Go 1.23.
```

Comment:
```suggestion
	// crypto/rand.Reader always fills the entire slice.
	// Since Go 1.24, it is guaranteed to never return an error, see https://github.com/golang/go/issues/66821.
	// This error check can be removed once we drop support for Go 1.23.
```